snuba/web/query.py

```diff
@@ -172,12 +172,14 @@ def _format_storage_query_and_run(
     ) as span:
         span.set_tag("table", source)
         try:
-            span.set_tag(
-                "ast_query",
-                AstSqlQuery(clickhouse_query, request_settings).format_sql(),
-            )
+            sql_data = AstSqlQuery(clickhouse_query, request_settings).sql_data()
+            span.set_tag("query_type", "ast")
         except Exception:
-            logger.warning("Failed to format ast query", exc_info=True)
+            sql_data = formatted_query.sql_data()
+            span.set_data("query_type", "dict")
+
+        for k, v in sql_data:
+            span.set_data(k, v)
```

Comment:
It would be interesting to see if the limit the sdk imposes to the data field is the same as the one imposed to the description (where the queries get cut today).
Could you please try to pass the entire query sql structure into one single call to set_data and see how it looks like for a huge query? I can show you offline where to find a big query from production.

Reply:
I attempted this, but it was ugly. With your suggestion to have both query types added, I'll bring this back.

Reply:
Pay attention, I think we still should figure out whether the query fits the field size limit by squeezing in a single field. Now we saw it is not the case very often when writing the formatted string (in the description field) but I am not sure the limit is the same for the `data` field. 
I can provide you offline a huge query to see where it fits.